flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/utils/TestRowDataCsvInputFormat.java

```diff
@@ -0,0 +1,179 @@
+		this.fieldNames = Arrays.asList(rowType.getFieldNames());
+
+		List<String> csvFieldNames = fieldNames.stream()
+				.filter(name -> !partitionKeys.contains(name)).collect(Collectors.toList());
+
+		List<String> selectFieldNames = Arrays.stream(selectFields)
+				.mapToObj(fieldNames::get)
+				.collect(Collectors.toList());
+		List<String> csvSelectFieldNames = selectFieldNames.stream()
+				.filter(name -> !partitionKeys.contains(name)).collect(Collectors.toList());
+		List<TypeInformation> csvSelectTypes = csvSelectFieldNames.stream()
+				.map(name -> fieldTypes.get(fieldNames.indexOf(name))).collect(Collectors.toList());
+		this.csvSelectConverters = csvSelectTypes.stream()
+				.map(TypeConversions::fromLegacyInfoToDataType)
+				.map(DataFormatConverters::getConverterForDataType)
+				.collect(Collectors.toList());
+		int[] csvSelectFields = csvSelectFieldNames.stream().mapToInt(csvFieldNames::indexOf).toArray();
+		this.inputFormat = new RowCsvInputFormat(
+				null, csvSelectTypes.toArray(new TypeInformation[0]), csvSelectFields);
+		this.inputFormat.setFilePaths(paths);
+
+		this.csvFieldMapping = csvSelectFieldNames.stream().mapToInt(selectFieldNames::indexOf).toArray();
+		this.emitted = 0;
+	}
+
+	@Override
+	public void configure(Configuration parameters) {
+		inputFormat.configure(parameters);
+	}
+
+	@Override
+	public BaseStatistics getStatistics(BaseStatistics cachedStatistics) throws IOException {
+		return inputFormat.getStatistics(cachedStatistics);
+	}
+
+	@Override
+	public FileInputSplit[] createInputSplits(int minNumSplits) throws IOException {
+		return inputFormat.createInputSplits(minNumSplits);
+	}
+
+	@Override
+	public InputSplitAssigner getInputSplitAssigner(FileInputSplit[] inputSplits) {
+		return inputFormat.getInputSplitAssigner(inputSplits);
+	}
+
+	@Override
+	public void open(FileInputSplit split) throws IOException {
+		inputFormat.open(split);
+		Path path = split.getPath();
+		LinkedHashMap<String, String> partSpec = PartitionPathUtils.extractPartitionSpecFromPath(path);
+		this.row = new GenericRow(selectFields.length);
+		for (int i = 0; i < selectFields.length; i++) {
+			int selectField = selectFields[i];
+			String name = fieldNames.get(selectField);
+			if (partitionKeys.contains(name)) {
+				String value = partSpec.get(name);
+				value = defaultPartValue.equals(value) ? null : value;
+				this.row.setField(
+						i, convertStringToInternal(value, fieldTypes.get(selectField)));
+			}
+		}
+		this.csvRow = new Row(csvSelectConverters.size());
+	}
+
+	private Object convertStringToInternal(String value, TypeInformation type) {
+		if (type.equals(Types.INT)) {
+			return Integer.parseInt(value);
+		} else if (type.equals(Types.LONG)) {
+			return Long.parseLong(value);
+		} else if (type.equals(Types.STRING)) {
+			return BinaryString.fromString(value);
+		} else {
+			throw new UnsupportedOperationException("Unsupported type: " + type);
```

Comment:
The `value` only from  partition column, use `Unsupported partition type` to make it clear?